<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1084</width>
    <height>728</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="project" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>20</y>
      <width>691</width>
      <height>621</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="Menu" native="true">
    <property name="geometry">
     <rect>
      <x>19</x>
      <y>19</y>
      <width>291</width>
      <height>661</height>
     </rect>
    </property>
    <widget class="QPushButton" name="open_button">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="text">
      <string>Open file</string>
     </property>
    </widget>
    <widget class="QGroupBox" name="Rotation">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>120</y>
       <width>121</width>
       <height>141</height>
      </rect>
     </property>
     <property name="title">
      <string>Rotation</string>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>101</width>
        <height>103</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="1">
        <widget class="QPushButton" name="y_rotate">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>33</height>
          </size>
         </property>
         <property name="text">
          <string>&gt;&gt;</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="x_rotate">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>33</height>
          </size>
         </property>
         <property name="text">
          <string>&gt;&gt;</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLineEdit" name="x_value_rot">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>20</height>
          </size>
         </property>
         <property name="text">
          <string>0</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLineEdit" name="y_value_rot">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>20</height>
          </size>
         </property>
         <property name="text">
          <string>0</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLineEdit" name="z_value_rot">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>20</height>
          </size>
         </property>
         <property name="text">
          <string>0</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="z_rotate">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>33</height>
          </size>
         </property>
         <property name="text">
          <string>&gt;&gt;</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QGroupBox" name="Transition">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>120</y>
       <width>121</width>
       <height>141</height>
      </rect>
     </property>
     <property name="title">
      <string>Transition</string>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>11</x>
        <y>30</y>
        <width>101</width>
        <height>103</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="1">
        <widget class="QPushButton" name="x_move">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>33</height>
          </size>
         </property>
         <property name="text">
          <string>&gt;&gt;</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLineEdit" name="y_value_tr">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>20</height>
          </size>
         </property>
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="z_move">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>33</height>
          </size>
         </property>
         <property name="text">
          <string>&gt;&gt;</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLineEdit" name="z_value_tr">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>20</height>
          </size>
         </property>
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLineEdit" name="x_value_tr">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>20</height>
          </size>
         </property>
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="y_move">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>33</height>
          </size>
         </property>
         <property name="text">
          <string>&gt;&gt;</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QGroupBox" name="Properties">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>350</y>
       <width>291</width>
       <height>251</height>
      </rect>
     </property>
     <property name="title">
      <string>Properties</string>
     </property>
     <widget class="QComboBox" name="vertex_type">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>110</y>
        <width>141</width>
        <height>33</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>33</height>
       </size>
      </property>
      <item>
       <property name="text">
        <string>square vertex</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>circle vertex</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>no vertex</string>
       </property>
      </item>
     </widget>
     <widget class="QPushButton" name="setBgColor">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>30</y>
        <width>121</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Background coror</string>
      </property>
     </widget>
     <widget class="QSlider" name="pointSize_slider">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>180</y>
        <width>191</width>
        <height>25</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>200</number>
      </property>
      <property name="pageStep">
       <number>1</number>
      </property>
      <property name="value">
       <number>30</number>
      </property>
      <property name="sliderPosition">
       <number>30</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QWidget" name="Projection" native="true">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>141</width>
        <height>41</height>
       </rect>
      </property>
      <widget class="QRadioButton" name="central">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>141</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Central projection</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="parallel">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>20</y>
         <width>141</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Parallel projection</string>
       </property>
      </widget>
     </widget>
     <widget class="QSlider" name="lineWidth_slider">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>150</y>
        <width>191</width>
        <height>25</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>200</number>
      </property>
      <property name="pageStep">
       <number>1</number>
      </property>
      <property name="value">
       <number>10</number>
      </property>
      <property name="sliderPosition">
       <number>10</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QComboBox" name="line_type">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>141</width>
        <height>33</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>33</height>
       </size>
      </property>
      <item>
       <property name="text">
        <string>solid line</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>dashed line</string>
       </property>
      </item>
     </widget>
     <widget class="QPushButton" name="setVertColor">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>110</y>
        <width>121</width>
        <height>33</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>33</height>
       </size>
      </property>
      <property name="text">
       <string>Vertex color</string>
      </property>
     </widget>
     <widget class="QPushButton" name="saveSettigs">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>210</y>
        <width>271</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Save Settings</string>
      </property>
     </widget>
     <widget class="QLabel" name="line_size_label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>150</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Line size</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="vert_size_label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>180</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Vertex size</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="setLineColor">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>80</y>
        <width>121</width>
        <height>33</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>33</height>
       </size>
      </property>
      <property name="text">
       <string>Line color</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="Scale">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>270</y>
       <width>161</width>
       <height>71</height>
      </rect>
     </property>
     <property name="title">
      <string>Scale</string>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>141</width>
        <height>51</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <widget class="QLineEdit" name="scale_value">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>20</height>
          </size>
         </property>
         <property name="text">
          <string>0</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="scale_apply">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>33</height>
          </size>
         </property>
         <property name="text">
          <string>&gt;&gt;</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QGroupBox" name="AxisBox">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>120</y>
       <width>31</width>
       <height>141</height>
      </rect>
     </property>
     <property name="title">
      <string>Axis</string>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>35</y>
        <width>31</width>
        <height>91</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="x_label_rot">
         <property name="text">
          <string>X</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="y_label_rot">
         <property name="text">
          <string>Y</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="z_label_rot">
         <property name="text">
          <string>Z</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QPushButton" name="scale_reset">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>310</y>
       <width>121</width>
       <height>33</height>
      </rect>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>33</height>
      </size>
     </property>
     <property name="text">
      <string>Normalize scale</string>
     </property>
    </widget>
    <widget class="QPushButton" name="transition_reset">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>280</y>
       <width>121</width>
       <height>33</height>
      </rect>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>33</height>
      </size>
     </property>
     <property name="text">
      <string>Move to center</string>
     </property>
    </widget>
    <widget class="QPushButton" name="transition_apply">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>670</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Move</string>
     </property>
    </widget>
    <widget class="QPushButton" name="loadSettings">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>0</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="text">
      <string>Load Settings</string>
     </property>
    </widget>
    <widget class="QGroupBox" name="Infobox">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>40</y>
       <width>291</width>
       <height>71</height>
      </rect>
     </property>
     <property name="title">
      <string/>
     </property>
     <widget class="QLabel" name="filename_value">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>6</y>
        <width>181</width>
        <height>21</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="filename_label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>6</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Filename:</string>
      </property>
     </widget>
     <widget class="QLabel" name="vertices_info_label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>26</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Vertices:</string>
      </property>
     </widget>
     <widget class="QLabel" name="vertices_value">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>26</y>
        <width>181</width>
        <height>21</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="edges_info_label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>46</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Edges:</string>
      </property>
     </widget>
     <widget class="QLabel" name="edges_value">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>46</y>
        <width>181</width>
        <height>21</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="ScreenBox">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>610</y>
       <width>291</width>
       <height>51</height>
      </rect>
     </property>
     <property name="title">
      <string>PrintScreen</string>
     </property>
     <widget class="QRadioButton" name="radio_gif">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>20</y>
        <width>41</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>.gif</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="radio_jpg">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>20</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>.jpg</string>
      </property>
     </widget>
     <widget class="QPushButton" name="save_button">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>20</y>
        <width>111</width>
        <height>31</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Save img</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="radio_bmp">
      <property name="geometry">
       <rect>
        <x>5</x>
        <y>20</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>.bmp</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="developer_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>690</y>
      <width>211</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string> 2023 hirokose, deepeste, mitzifra</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="full_filename_label">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>690</y>
      <width>631</width>
      <height>20</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: blue;</string>
    </property>
    <property name="text">
     <string>filename link</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QSlider" name="y_slider">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>660</y>
      <width>311</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-25</number>
    </property>
    <property name="maximum">
     <number>25</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="sliderPosition">
     <number>0</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="x_slider">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>390</y>
      <width>22</width>
      <height>251</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: red;</string>
    </property>
    <property name="minimum">
     <number>-25</number>
    </property>
    <property name="maximum">
     <number>25</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QSlider" name="z_slider">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>660</y>
      <width>301</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-25</number>
    </property>
    <property name="maximum">
     <number>25</number>
    </property>
    <property name="pageStep">
     <number>5</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
    <property name="sliderPosition">
     <number>0</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="X_label">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>360</y>
      <width>21</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="Y_label">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>660</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="Z_label">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>660</y>
      <width>21</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>project</class>
   <extends>QWidget</extends>
   <header>project.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
